src/error.rs

```diff
@@ -1,9 +1,6 @@
 /// The error type used by the `Routerify` library.
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
```

Comment:
Is there a reason to keep two different error types? I was trying to write a comment update to explain `Error` and `RouterError` and when a user should use / expect one over the other. 

I realize you don't have to parameterize the variants in this Error on `<E>` but I wonder if the whole crate would be simpler if you simply put all the variants into `RouterError`

Reply:
The aim was to avoid type-parameterising things on E that aren't already, and arguably don't make sense to be given they can never return such an error type. For example, [ScopedDataMap](https://github.com/influxdata/routerify/blob/propagate-custom-errors/src/data_map/scoped.rs) or the [regex helpers](https://github.com/influxdata/routerify/blob/propagate-custom-errors/src/regex_generator/mod.rs#L38).

Reply:
I've made some further changes that constrain `RouterError` to only be used within Router and its corresponding hyper Service's - it is no longer used within the route nor middleware layers